File: demo/src/main/java/com/baozi/demo/fragment/SimpleRecyclerViewFg.java
package com.baozi.demo.fragment;

import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.baozi.demo.R;
import com.baozi.treerecyclerview.adpater.TreeRecyclerAdapter;
import com.baozi.treerecyclerview.base.BaseRecyclerAdapter;


public abstract class SimpleRecyclerViewFg<T extends BaseRecyclerAdapter> extends Fragment {
    protected RecyclerView recyclerView;
    protected Context context;
    protected TreeRecyclerAdapter adapter = new TreeRecyclerAdapter();

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.context = context;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        if (recyclerView == null) {
            recyclerView = (RecyclerView) inflater.inflate(R.layout.layout_rv_content, container, false);
            recyclerView.setBackgroundColor(Color.GRAY);
        }
        return recyclerView;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setLayoutManager(new LinearLayoutManager(context));
        recyclerView.setAdapter(adapter);
    }
}

File: demo/src/main/java/com/baozi/demo/fragment/SimpleRecyclerViewFg.kt
package com.baozi.demo.fragment

import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.baozi.demo.R
import com.baozi.treerecyclerview.adpater.TreeRecyclerAdapter
import com.baozi.treerecyclerview.base.BaseRecyclerAdapter


abstract class SimpleRecyclerViewFg<T : BaseRecyclerAdapter<*>> : Fragment() {
    protected var recyclerView: RecyclerView? = null
    protected var mContext: Context? = null
    protected open var adapter = TreeRecyclerAdapter()

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        this.mContext = context
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (recyclerView == null) {
            recyclerView = inflater.inflate(R.layout.layout_rv_content, container, false) as RecyclerView
            recyclerView!!.setBackgroundColor(Color.GRAY)
        }
        return recyclerView
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        recyclerView!!.itemAnimator = DefaultItemAnimator()
        recyclerView!!.layoutManager = LinearLayoutManager(context)
        recyclerView!!.adapter = adapter
    }
}
